core/domain/suggestion_services.py

```diff
@@ -288,17 +308,47 @@ def reject_suggestion(suggestion, reviewer_id, review_message):
     Raises:
         Exception: The suggestion is already handled.
     """
-    if suggestion.is_handled:
-        raise Exception('The suggestion has already been accepted/rejected.')
+
+    reject_suggestions([suggestion], reviewer_id, review_message)
+
+
+def reject_suggestions(suggestions, reviewer_id, review_message):
+    """Rejects the suggestions.
+
+    Args:
+        suggestions: list(Suggestion). The suggestions to be rejected.
+        reviewer_id: str. The ID of the reviewer rejecting the suggestions.
+        review_message: str. The message provided by the reviewer while
+            rejecting the suggestions.
+
+    Raises:
+        Exception: One of the suggestions has already been handled.
+    """
+
+    for suggestion in suggestions:
+        if suggestion.is_handled:
+            raise Exception(
+                'The suggestion has already been accepted/rejected.')
     if not review_message:
         raise Exception('Review message cannot be empty.')
-    mark_review_completed(
-        suggestion, suggestion_models.STATUS_REJECTED, reviewer_id)
 
-    thread_id = suggestion.suggestion_id
-    feedback_services.create_message(
-        thread_id, reviewer_id, feedback_models.STATUS_CHOICES_IGNORED,
-        None, review_message)
+    suggestion_ids = [suggestion.suggestion_id for suggestion in suggestions]
+    suggestion_models_to_reject = (
+        suggestion_models.GeneralSuggestionModel.get_multi(suggestion_ids)
+    )
+
+    for suggestion_model in suggestion_models_to_reject:
+        suggestion_model.status = suggestion_models.STATUS_REJECTED
+        suggestion_model.final_reviewer_id = reviewer_id
+
+    suggestion_models.GeneralSuggestionModel.put_multi(
+        suggestion_models_to_reject)
+
+    for suggestion in suggestions:
+        thread_id = suggestion.suggestion_id
+        feedback_services.create_message(
```

Comment:
Look into making a create_multi() version of this so that you don't do N individual put() calls.